[build] Don't include GraphicsContextGL.h in WebCoreArgumentCoders.h

djg opened the pull request:
#### df7bdf36f1cca4ea2e146d616a0cb821d98f82f5
<pre>
[build] Don&apos;t include GraphicsContextGL.h in WebCoreArgumentCoders.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=251238">https://bugs.webkit.org/show_bug.cgi?id=251238</a>
rdar://problem/104722312

Reviewed by Dean Jackson.

WebCoreArgumentCoders only needs a few simple declarations from
GraphicsContextGL.h. Refactor GraphicsContextGL.h to extra enum
SimulatedEventForTesting and struct GraphicsContextGLActiveInfo, allowing the
required pieces to be included with out need to pull in the entire
expensive-to-parse header.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/GraphicsContextGL.h:
* Source/WebCore/platform/graphics/GraphicsContextGLActiveInfo.h: Added.
* Source/WebCore/platform/graphics/GraphicsContextGLEnums.h: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:

Canonical link: <a href="https://commits.webkit.org/259605@main">https://commits.webkit.org/259605@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed6c8d2c08bb820725ee88daa8800c1b807d6350

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105398 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14467 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38271 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114654 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174821 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109301 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15659 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5404 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97702 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114537 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111156 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12097 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95080 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39580 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93963 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26715 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81266 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7781 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28074 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7901 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13924 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47619 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6623 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9692 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/platform/graphics/GraphicsContextGL.h:
```diff
@@ -28,7 +28,9 @@
 #if ENABLE(WEBGL)
 
 #include "DestinationColorSpace.h"
+#include "GraphicsContextGLActiveInfo.h"
```
It looks like GraphicsContextGLActiveInfo can get forward declared rather than included here.

djg replied:
I've included it here to make `GeneratedSerializers.mm` compile because I didn't want to learn how `GeneratedSerializers.mm` is generated. I've investigated the generator and implemented it properly now.

jernoble reviewed (commented):
r=me with one nit. It might create a lot more work as we'd have to add more include directives in files that include GraphicsContextGL.h, so I'm marking it approved regardless.

djg committed:
[build] Don't include GraphicsContextGL.h in WebCoreArgumentCoders.h
https://bugs.webkit.org/show_bug.cgi?id=251238
rdar://problem/104722312

Reviewed by Dean Jackson.

WebCoreArgumentCoders only needs a few simple declarations from
GraphicsContextGL.h. Refactor GraphicsContextGL.h to extra enum
SimulatedEventForTesting and struct GraphicsContextGLActiveInfo, allowing the
required pieces to be included with out need to pull in the entire
expensive-to-parse header.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/GraphicsContextGL.h:
* Source/WebCore/platform/graphics/GraphicsContextGLActiveInfo.h: Added.
* Source/WebCore/platform/graphics/GraphicsContextGLEnums.h: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.h:

Canonical link: https://commits.webkit.org/259605@main
